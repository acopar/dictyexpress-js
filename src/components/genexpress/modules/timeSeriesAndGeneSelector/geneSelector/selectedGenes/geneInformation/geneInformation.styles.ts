import styled from 'styled-components';

export const GeneInformationContainer = styled.div`
    min-width: 240px;
    max-width: 300px;
    border: 1px solid;
    padding: ${(props): string => `${props.theme.spacing(1)}px`};
    background-color: ${(props): string => props.theme.palette.background.paper};
`;

export const GeneInformationHeader = styled.div`
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    align-content: center;
    margin-bottom: 10px;
`;

export const GeneInformationTitle = styled.h3`
    margin: 0;
    display: flex;
    align-self: center;
`;

export const GeneInformationLabel = styled.div`
    color: #bdbdbd;
    margin-bottom: 5px;
`;

export const GeneInformationValue = styled.div`
    margin-bottom: 10px;
`;

export const DictyBaseLogo = styled.img`
    height: 25px;
`;
