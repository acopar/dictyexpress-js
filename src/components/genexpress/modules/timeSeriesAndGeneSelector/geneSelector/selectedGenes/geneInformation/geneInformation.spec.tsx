import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { customRender } from 'tests/test-utils';
import { generateGene } from 'tests/mock';
import { GeneInformation } from './geneInformation';

const gene = generateGene(0);

describe('geneInformation', () => {
    it('should render as default snapshot', () => {
        const { asFragment } = customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should show Unhighlight button if gene is highlighted', () => {
        customRender(
            <GeneInformation
                gene={gene}
                highlighted
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        screen.getByText('Unhighlight');
    });

    it('should show Unhighlight button if gene is highlighted', () => {
        customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        screen.getByText('Highlight');
    });

    it('should call onHighlight when user clicks Highlight', () => {
        const mockedOnHighlight = jest.fn();
        customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={mockedOnHighlight}
                onUnhighlight={(): void => {}}
            />,
        );

        fireEvent.click(screen.getByText('Highlight'));

        expect(mockedOnHighlight).toBeCalled();
    });

    it('should call onUnhighlight when user clicks Unhighlight', () => {
        const mockedOnUnhighlight = jest.fn();
        customRender(
            <GeneInformation
                gene={gene}
                highlighted
                onHighlight={(): void => {}}
                onUnhighlight={mockedOnUnhighlight}
            />,
        );

        fireEvent.click(screen.getByText('Unhighlight'));

        expect(mockedOnUnhighlight).toBeCalled();
    });
});
