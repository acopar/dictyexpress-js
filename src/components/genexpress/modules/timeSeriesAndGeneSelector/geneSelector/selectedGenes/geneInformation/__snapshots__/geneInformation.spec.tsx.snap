// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`geneInformation should render as default snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe jNOcwz"
  >
    <div
      class="sc-gswNZR fJgLXn"
    >
      <h3
        class="sc-dkrFOg jxeFeE"
      >
        Gene information
      </h3>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-bl1inz-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Highlight
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <p>
      TestGene0
    </p>
    <div>
      <div
        class="sc-hLBbgP jSEqQq"
      >
        Gene ID:
      </div>
      <div
        class="sc-eDvSVe mGJtp"
      >
        0
      </div>
    </div>
    <div>
      <div
        class="sc-hLBbgP jSEqQq"
      >
        Alternatives:
      </div>
      <div
        class="sc-eDvSVe mGJtp"
      >
        TestGene0_alias1, TestGene0_alias2
      </div>
    </div>
    <div>
      <div
        class="sc-hLBbgP jSEqQq"
      >
        Database link:
      </div>
      <div
        class="sc-eDvSVe mGJtp"
      >
        <a
          href="http://dictybase.org/id/0"
          rel="noopener noreferrer"
          target="_blank"
        >
          Open in dictyBase
          <img
            alt="Open in dictyBase."
            class="sc-jSUZER jYafAW"
            src="dictybase_logo2.jpg"
          />
        </a>
      </div>
    </div>
    <div>
      <div
        class="sc-hLBbgP jSEqQq"
      >
        Description:
      </div>
      <div
        class="sc-eDvSVe mGJtp"
      >
        This is a test gene 0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
